octopusdeploy/schema_amazon_web_services_openid_connect_account.go

```diff
@@ -0,0 +1,130 @@
+
+	if v, ok := d.GetOk("description"); ok {
+		account.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("environments"); ok {
+		account.EnvironmentIDs = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("space_id"); ok {
+		account.SpaceID = v.(string)
+	}
+
+	if v, ok := d.GetOk("tenanted_deployment_participation"); ok {
+		account.TenantedDeploymentMode = core.TenantedDeploymentMode(v.(string))
+	}
+
+	if v, ok := d.GetOk("tenant_tags"); ok {
+		account.TenantTags = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("tenants"); ok {
+		account.TenantIDs = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("execution_subject_keys"); ok {
+		account.DeploymentSubjectKeys = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("health_subject_keys"); ok {
+		account.HealthCheckSubjectKeys = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("account_test_subject_keys"); ok {
+		account.AccountTestSubjectKeys = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("session_duration"); ok {
+		account.SessionDuration = v.(string)
+	}
+
+	return account
+}
+
+func getAmazonWebServicesOpenIDConnectAccountSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"description": {
+			Description: "A user-friendly description of this AWS OIDC account.",
+			Optional:    true,
+			Type:        schema.TypeString,
+		},
+		"environments": getEnvironmentsSchema(),
+		"name": {
+			Description:      "The name of this AWS OIDC account.",
+			Required:         true,
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(1, 200)),
+		},
+		"space_id":                          getSpaceIDSchema(),
+		"tenanted_deployment_participation": getTenantedDeploymentSchema(),
+		"tenants":                           getTenantsSchema(),
+		"tenant_tags":                       getTenantTagsSchema(),
+		"execution_subject_keys":            getSubjectKeysSchema(SchemaSubjectKeysDescriptionExecution),
+		"health_subject_keys":               getSubjectKeysSchema(SchemaSubjectKeysDescriptionHealth),
+		"account_test_subject_keys":         getSubjectKeysSchema(SchemaSubjectKeysDescriptionAccountTest),
+		"role_arn": {
+			Description: "The Amazon Resource Name (ARN) of the role that the caller is assuming.",
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"session_duration": {
+			Description: "The duration, in seconds, of the role session.",
+			Required:    false,
+			Optional:    true,
+			Type:        schema.TypeInt,
+		},
+	}
+}
+
+func setAmazonWebServicesOpenIDConnectAccount(ctx context.Context, d *schema.ResourceData, account *accounts.AwsOIDCAccount) error {
+	d.Set("description", account.GetDescription())
+	d.Set("name", account.GetName())
+	d.Set("space_id", account.GetSpaceID())
+	d.Set("tenanted_deployment_participation", account.GetTenantedDeploymentMode())
+	d.Set("role_arn", account.RoleArn)
```

Comment:
Should this include session_duration and audience?

Reply:
fixed `session_duration` good catch.
the aws oidc account doesn't have an `audience`